### PR TITLE
chores: move `role_playing_with_functions` from utils into example

### PR DESCRIPTION
**Pull request by Wendong-Fan**
## Description

move role_playing_with_functions from utils into example 
## Motivation and Context

Why is this change required? What problem does it solve?

- [ ] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ ] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Wendong-Fan** [comment]
> I've reviewed all the code in this PR, and the changes are quite minimal, mainly involving some reorganization of call paths. I believe there are essentially no problems.
> 
> There's just one point I might need your clarification on. If I remember correctly, `role_playing_with_functions` was initially in the `example` directory. Later, the developers of Camel moved it to `utils`, and now it's been moved back. Could you explain the reason for this?

Hey @yiyiyi0817 , thanks for your review, after discussion with Guohao we decided to move this back, refer to https://github.com/camel-ai/camel/pull/472#discussion_r1593930095

**Review by yiyiyi0817 — APPROVED** [review]
OK, I understand. Thank you for contributing to it, LGTM.  @Wendong-Fan

File: examples/function_call/github_examples.py [hunk 1]
```diff
@@ -46,15 +46,15 @@ def solve_issue(
...
+        temperature=0.0,
     )
     agent = ChatAgent(
         assistant_sys_msg,
         model_type=model,
         model_config=assistant_model_config,
-        function_list=toolkit.get_tools(),
+        tools=toolkit.get_tools(),
     )
     agent.reset()
```
Comment (Appointat) [inline examples/function_call/github_examples.py:59]: Is the agent initialized, is `agent.reset()` necessary?

**Review by Appointat — COMMENTED** [review]
Reviewed, looks good to me.

**Pull request merged**